File: 0529/캐시/캐시_김태진.java
import java.util.ArrayList;

class Solution {
    public int solution(int cacheSize, String[] cities) {
        int answer = 0;
        if(cacheSize == 0)
            return cities.length * 5;
        ArrayList<String> s = new ArrayList<>();
        for(int i = 0; i < cities.length; i++){
            int j;
            cities[i] = cities[i].toLowerCase();
            for(j = 0; j < s.size(); j++){
                if(cities[i].equals(s.get(j))){
                    s.remove(j);
                    s.add(cities[i]);
                    answer++;
                    break;
                }
            }
            if(j == s.size()){
                answer += 5;
                if(s.size() >= cacheSize)
                    s.remove(0);
                s.add(cities[i]);
            }
        }
        return answer;
    }
}

File: 0529/캐시/캐시_김태진.cpp
#include <string>
#include <vector>
#include <algorithm>

using namespace std;

int solution(int cacheSize, vector<string> cities) {
    int answer = 0;
    if(cacheSize == 0)
        return cities.size() * 5;
    vector<string> v;
    for(int i = 0; i < cities.size(); i++){
        int j;
        transform(cities[i].begin(), cities[i].end(), cities[i].begin(), ::tolower);
        for(j = 0; j < v.size(); j++){
            if(cities[i] == v[j]){
                v.erase(v.begin() + j);
                v.push_back(cities[i]);
                answer++;
                break;
            }
        }
        if(j == v.size()){
            answer += 5;
            if(v.size() >= cacheSize)
                v.erase(v.begin());
            v.push_back(cities[i]);
        }
    }
    return answer;
}